conjure-java-core/src/integrationInput/java/com/palantir/product/MultipleOrderedStages.java

```diff
@@ -0,0 +1,224 @@
+        this.token = token;
+        this.item = item;
+        this.items = Collections.unmodifiableSet(items);
+        this.mappedRids = Collections.unmodifiableMap(mappedRids);
+    }
+
+    @JsonProperty("token")
+    public OneField getToken() {
+        return this.token;
+    }
+
+    @JsonProperty("item")
+    public String getItem() {
+        return this.item;
+    }
+
+    @JsonProperty("items")
+    public Set<SafeLong> getItems() {
+        return this.items;
+    }
+
+    @JsonProperty("mappedRids")
+    public Map<ResourceIdentifier, String> getMappedRids() {
+        return this.mappedRids;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return this == other || (other instanceof MultipleOrderedStages && equalTo((MultipleOrderedStages) other));
+    }
+
+    private boolean equalTo(MultipleOrderedStages other) {
+        return this.token.equals(other.token)
+                && this.item.equals(other.item)
+                && this.items.equals(other.items)
+                && this.mappedRids.equals(other.mappedRids);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = memoizedHashCode;
+        if (result == 0) {
+            result = Objects.hash(this.token, this.item, this.items, this.mappedRids);
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "MultipleOrderedStages{token: " + token + ", item: " + item + ", items: " + items + ", mappedRids: "
+                + mappedRids + '}';
+    }
+
+    private static void validateFields(
+            OneField token, String item, Set<SafeLong> items, Map<ResourceIdentifier, String> mappedRids) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, token, "token");
+        missingFields = addFieldIfMissing(missingFields, item, "item");
+        missingFields = addFieldIfMissing(missingFields, items, "items");
+        missingFields = addFieldIfMissing(missingFields, mappedRids, "mappedRids");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(4);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static ItemStageBuilder builder() {
+        return new Builder();
+    }
+
+    public interface ItemStageBuilder {
+        TokenStageBuilder item(@Nonnull String item);
+
+        Completed_StageBuilder from(MultipleOrderedStages other);
```

Comment:
I don't think this is correct, it doesn't allow any required (non-collection) fields to be modified.

The `Builder` type should be renamed to something like `DefaultBuilder` and made private, while this method returns a `Builder` interface which extends all the stages, allowing all methods to be called.

```java
interface ItemStageBuilder {
    TokenStageBuilder item(@Nonnull String item);
    Builder from(MultipleOrderedStages other);
}

public interface TokenStageBuilder {
    Completed_StageBuilder token(@Nonnull OneField token);
}

interface Completed_StageBuilder {
   MultipleOrderedStages build();
   /*etc*/
}

// new one:
interface Builder extends ItemStageBuilder, TokenStageBuilder, Completed_StageBuilder {
    @Override
    Builder item(@Nonnull String item);
    @Override
    Builder token(@Nonnull OneField token);
    /* etc, all builder methods are overridden to return this Builder type */
}

private static final class DefaultBuilder implements Builder {
   /* implementation */
}
````

Reply:
I like the idea just because i think this looks cleaner.

You'll notice however that in my current implementation, in the final stage, i also provide methods to set any of the required fields, therefore effectively allowing folks that used the `from` method to override any of those fields to whatever they want.

Reply:
Ah, I did not notice that! I think I'd prefer if the final stage didn't expose fields that we've already set in the purely functional workflow since it reduces complexity. What do you think?

Reply:
I agree with that! Let me adjust my implementation to match your example :)